src/cart/crt_rpc.c

```diff
@@ -661,6 +664,34 @@ uri_lookup_cb(const struct crt_cb_info *cb_info)
 		D_GOTO(out, rc);
 	}
 
+	/* After a URI lookup, check if membership list has this rank.
+	 * If not - we discovered a new rank and need to populate it in membs
+	 * list of the group.
+	 */
+	D_RWLOCK_WRLOCK(&default_grp_priv->gp_rwlock);
+	membs = grp_priv_get_membs(grp_priv);
+	found = false;
+
+	if (membs) {
+		for (i = 0; i < membs->rl_nr; i++) {
+			if (membs->rl_ranks[i] == ul_in->ul_rank) {
+				found = true;
+				break;
+			}
+		}
+	}
```

Comment:
could this be condensed to the following single line?
found = d_rank_list_find(membs, ul_in->ul_rank, NULL):

Reply:
indeed, will update once builds stabilize